### DISCUSSION TITLE
Help installing helloSystem

### DISCUSSION THREAD

**Discussion by SanPollo** [helloSystem/hello#317] [Q&A] [answered]
Hi there,

Just to give you some background about myself; I have been using FreeBSD for around 13 years now, but haven't done anything GUI-related for around 10 years when I migrated to Mac OS X (by way of hackintosh - I couldn't afford the hardware, but liked the UI and the underlying UNIX). Around the same time, I got a job as an engineer at the Apple Premium Reseller in my hometown where I did hardware repairs, and Mac support, for around 3 years. After leaving there I was mainly using Windows with FreeBSD servers.

Fast forward to now, Microsoft no longer want my netbook to run Windows 10 at a reasonable speed, so I installed FreeBSD on it (which I love). But I'm having trouble choosing a window manager. I tried Lumina as I believed it was designed to be light and designed for FreeBSD, and found it to be quite clunky and out of date. I also tried MATE which looked exactly like Gnome I used to use 10 years ago. It's okay but I don't really like the look and feel. Finally I looked at some of the lighter window managers like cwm. I quite liked the screenshots of cwm on the net, but after installing it I realised I would have to do all of the customisations manually myself.

So I thought I would try helloSystem as it is based on my favourite OS, and has a Mac look and feel to it. Unfortunately I have a slight problem in that the USB boot image requires 4GB RAM and my netbook only has 2GB RAM so I have a few questions:

1. How can I install helloSystem onto my 2GB machine? Is the only option to put the SSD in another machine that has 4GB RAM and to install it that way? Or would that cause issues if the hardware of the other machine is significantly different?

2. I tried NomadBSD on the same machine, with which I was very impressed. Literally everything worked on the laptop including the webcam and built in sound, and it didn't require 4GB of RAM to decompress an ISO into. Is there any way a live USB of helloSystem could be made available in future which isn't a compressed ISO image so that people like myself could try out, and install it if we liked it? I think this would encourage more people to give helloSystem a go, especially on older hardware.

Finally I just wanted to say thank you to everyone who has put effort into putting helloSystem together. I really hope it takes off because, as a FreeBSD and a Mac fan, I just want to install an OS on my old laptop and for it to "just work" 😃 

Thanks in advance for any help,
Nick

**Comment by SanPollo** [comment]
Okay so I've done some digging and it seems that the [documentation that mentions 4GB](https://hellosystem.github.io/docs/developer/building.html) is out of date, as is the linked page [hello Live ISO builder](https://github.com/helloSystem/ISO) which says that 0.6.0 is the latest version. Can I volunteer to write / update documentation? I have [started a wiki](https://wiki.bromania.eu/) with some FreeBSD guides on it if you would like to see what my instruction-writing is like.

Meanwhile, I have installed 0.7.0 on my netbook. It has an Atheros wifi card installed in it, which works fine with both standard FreeBSD (by manually configuring /etc/wpa_supplicant.conf) and NomadBSD (in the UI). However in helloSystem it doesn't see any networks.

I tried copying the information from the wpa_supplicant.conf in the FreeBSD installation and rebooted, but helloSystem now hangs on the boot screen. Any ideas regarding this?

**Comment by probonopd** [comment, marked as answer]
Hello @SanPollo, and thanks for sharing your thoughts.

> Unfortunately I have a slight problem in that the USB boot image requires 4GB RAM and my netbook only has 2GB RAM 

That requirement was reduced in helloSystem 0.7.0 which should be able to boot on 2 GB RAM machines. Starting with this version, the whole Live boot system was reworked entirely. The compressed filesystem is no longer unpacked to RAM at boot time, resulting in significantly lower RAM consumption and faster boot times.

Running a web browser on a 2 GB RAM machine will not be fun, though - not on helloSystem and probably not on most other operating systems. Most likely you know have experienced this already. I don't really know what has made browsers so bloated as of late but this is my experience unfortunately.

> Okay so I've done some digging and it seems that the documentation that mentions 4GB is out of date, as is the linked page hello Live ISO builder which says that 0.6.0 is the latest version.

You have found a piece of outdated information.

> Can I volunteer to write / update documentation?

Absolutely, this would be a highly valued contribution. Please do send individual GitHub Pull Requests for things like the outdated 0.6.0 reference. Also, please have a look at https://hellosystem.github.io/docs/. (The https://hellosystem.github.io/docs/developer/boot page is currently not updated yet to reflect 0.7.0 and possibly there are other instances as well.)

> Meanwhile, I have installed 0.7.0 on my netbook. It has an Atheros wifi card installed in it, which works fine with both standard FreeBSD (by manually configuring /etc/wpa_supplicant.conf) and NomadBSD (in the UI). However in helloSystem it doesn't see any networks.

Please see https://github.com/helloSystem/ISO/issues/330 and test whether the workarounds discussed there help on the helloSystem Live ISO. Please do [open a new ticket](https://github.com/helloSystem/ISO/issues/new?assignees=&labels=bug&template=bug_report.md&title=) describing your exact Atheros hardware, config, etc. If some hardware works on any FreeBSD based system but not on helloSystem, then I'd consider it a bug ;-)

> I tried copying the information from the wpa_supplicant.conf in the FreeBSD installation and rebooted, but helloSystem now hangs on the boot screen. Any ideas regarding this?

Please do a [verbose boot](https://hellosystem.github.io/docs/developer/boot.html#boot-in-verbose-mode) and see whether that gives any clues.

Again, welcome to helloSystem!

**Reply by SanPollo** [reply]
Many thanks for your reply @probonopd and for the warm welcome.

I had a chance to play around with helloSystem for a couple of hours today, and have been jotting down any potential bugs and suggestions for improvement. I installed the 0.8.0 build in the end (to avoid logging defects if something has already been fixed). I work as a software support engineer, and am used to logging improvements and defects in Jira, but I have never really used github before (other than to download pre-existing software to compile) so apologies if I do something wrong - I had to look up the difference between a pull request and an issue when you mentioned it! 😀 

I will try the web browser and see what happens, but I agree that modern browsers are bloated. Everyone seems to want web apps these days which, I guess, is part of the cloud trend.

Thanks for the link to the issue regarding the Atheros card. I booted into single user mode to revert the changes I did to the wpa_supplicant.conf and it booted fine after that (albeit still with no wifi). I've backed the netbook away for the night, but will try the workarounds in the issue tomorrow. The sooner I can get wifi up and running, the sooner I can take screenshots to accompany the issues I report!
